plugins/inputs/ethtool/ethtool_linux.go

```diff
@@ -0,0 +1,116 @@
+// +build linux
+
+package ethtool
+
+import (
+	"github.com/influxdata/telegraf/filter"
+	"github.com/pkg/errors"
+	"net"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+	"github.com/safchain/ethtool"
+)
+
+type CommandEthtool struct {
+	ethtool *ethtool.Ethtool
+}
+
+func (e *Ethtool) Gather(acc telegraf.Accumulator) error {
+
+	// Get the list of interfaces
+	interfaces, err := e.command.Interfaces()
+	if err != nil {
+		acc.AddError(err)
+		return nil
+	}
+
+	interfaceFilter, err := filter.NewIncludeExcludeFilter(e.InterfaceInclude, e.InterfaceExclude)
+	if err != nil {
+		return err
+	}
+
+	for _, iface := range interfaces {
+
+		// Check this isn't a loop back and that its matched by the filter
+		if (iface.Flags&net.FlagLoopback == 0) && interfaceFilter.Match(iface.Name) {
+			e.wg.Add(1)
+			go e.gatherEthtoolStats(iface, acc)
```

Comment:
Lets localize the `wg` `Done` call. Can you put call this in an anonymous function:
```go
go func(iface whatever) {
  e.gatherEthtoolStats(iface, acc)
  e.wg.Done()
}(iface)
```

This will also allow you to initialize it here and remove the `wg` from the struct.